calico_containers/pycalico/datastore.py

```diff
@@ -291,9 +292,31 @@ def ensure_global_config(self):
         # Disable felix status reporting, which consumes etcd write bandwidth.
         self._write_global_config(FELIX_REPORTING_INTERVAL_PATH, "0")
 
+        # Create our Cluster GUID (if it does not already exist).
+        self._ensure_cluster_guid(CLUSTER_GUID_PATH)
```

Comment:
Why not just use _write_global_config() here rather than implementing a new method to do the same thing.  It means you have to create a UUID regardless of whether you need to, but that shouldn't be much of an overhead for start of day.